src/controllers/resource_buy.py

```diff
@@ -0,0 +1,78 @@
+from src.decorators.connect_to_store import connect
+
+
+@connect
+def buy_resource(request, **kwargs):
+    """Handle how a player can buy resources"""
+    get_state = kwargs['get_state']
+    dispatch = kwargs['dispatch']
+
+    # dict representing what the player is buying
+    resources_to_buy = request.get('resources')
+    player_id = request.get('player_id')
+    state = get_state()
+
+    # Get the player
+    player = [pl for pl in state.get(
+        'players') if pl.get('player_id') == player_id]
+    if not player:
+        return False
```

Comment:
Is this not handled by the @is_player_turn decorator?

Reply:
Oh yeah, that's true